frontend/public/src/containers/pages/DashboardPage.js

```diff
@@ -58,270 +48,85 @@ type DashboardPageProps = {
+}
+
 type DashboardPageState = {
-  submittingEnrollmentId: number | null,
-  activeMenuIds: number[],
-  emailSettingsModalVisibility: boolean[],
   programDrawerVisibility: boolean,
-  programDrawerEnrollments: any[],
+  programDrawerEnrollments: ?any[],
+  currentTab: string,
 }
 
 export class DashboardPage extends React.Component<
   DashboardPageProps,
   DashboardPageState
 > {
   state = {
-    submittingEnrollmentId:       null,
-    activeMenuIds:                [],
-    emailSettingsModalVisibility: [],
     programDrawerVisibility:      false,
-    programDrawerEnrollments:     [],
+    programDrawerEnrollments:     null,
+    currentTab:                   DashboardTab.courses,
   }
 
-  toggleEmailSettingsModalVisibility = (enrollmentId: number) => {
-    const { emailSettingsModalVisibility } = this.state
-    let isOpen = false
-    if (emailSettingsModalVisibility[enrollmentId] === undefined) {
-      isOpen = true
-    } else {
-      isOpen = !emailSettingsModalVisibility[enrollmentId]
-    }
-    emailSettingsModalVisibility[enrollmentId] = isOpen
+  toggleDrawer(enrollment: any) {
     this.setState({
-      emailSettingsModalVisibility: emailSettingsModalVisibility
-    })
-  }
-
-  isActiveMenuId(itemId: number): boolean {
-    return !!this.state.activeMenuIds.find(id => id === itemId)
-  }
-
-  toggleActiveMenuId(itemId: number) {
-    return () => {
-      const isActive = this.isActiveMenuId(itemId)
-      this.setState({
-        activeMenuIds: isActive
-          ? without([itemId], this.state.activeMenuIds)
-          : [...this.state.activeMenuIds, itemId]
-      })
-    }
-  }
-
-  toggleDrawer(enrollments: any) {
-    this.setState({
-      programDrawerEnrollments: enrollments,
+      programDrawerEnrollments: enrollment,
       programDrawerVisibility:  !this.state.programDrawerVisibility
     })
   }
 
-  async onDeactivate(enrollment: RunEnrollment) {
-    const { deactivateEnrollment, addUserNotification } = this.props
-    this.setState({ submittingEnrollmentId: enrollment.id })
-    try {
-      const resp = await deactivateEnrollment(enrollment.id)
-      let userMessage, messageType
-      if (isSuccessResponse(resp)) {
-        messageType = ALERT_TYPE_SUCCESS
-        userMessage = `You have been successfully unenrolled from ${enrollment.run.title}.`
-      } else {
-        messageType = ALERT_TYPE_DANGER
-        userMessage = `Something went wrong with your request to unenroll. Please contact support at ${SETTINGS.support_email}.`
-      }
-      addUserNotification({
-        "unenroll-status": {
-          type:  messageType,
-          props: {
-            text: userMessage
-          }
-        }
-      })
-      // Scroll to the top of the page to make sure the user sees the message
-      window.scrollTo(0, 0)
-    } finally {
-      this.setState({ submittingEnrollmentId: null })
+  toggleTab(tab: string) {
+    if (tab === DashboardTab.courses || tab === DashboardTab.programs) {
```

Comment:
Is this just for integrity or can this be removed?

Reply:
Integrity, yes.